src/LinAlg/hiopVectorRajaPar.hpp

```diff
@@ -54,9 +54,32 @@
  * @author Cosmin G. Petra <petra1@llnl.gov>, LLNL
  *
  */
+
+#include "hiopVectorRaja.hpp"
+
 #ifndef HIOP_VECTOR_RAJA_PAR
 #define HIOP_VECTOR_RAJA_PAR
 
+#ifdef HIOP_USE_RAJA
+namespace hiop
+{
+#ifdef HIOP_USE_CUDA
+  typedef hiopVectorRaja<MemBackendUmpire, ExecPolicyRajaCuda> hiopVectorRajaPar;
+#endif
+
+#ifdef HIOP_USE_HIP
+  typedef hiopVectorRaja<MemBackendUmpire, ExecPolicyRajaHip> hiopVectorRajaPar;
+#endif
+
+//todo: we need a macro for openmp raja
```

Comment:
If a user compiles with RAJA~cuda~hip~OpenMP, then they could shoot themselves in the foot. I think we should support sequential policy out of the box, but from discussion perhaps this is fine

Reply:
Next version of RAJA will fix the issue when HIP and OpenMP backends are enabled at the same time, so this issue will become moot. However, it is a good idea to have option to build RAJA with sequential policy, as well.

CC @davidbeckingsale

Reply:
I think I understand what you guys are alluding to. But RAJA with sequential policy does not make sense. You mean with OpenMP policy?

Reply:
RAJA sequential policy makes sense for users that are experimenting, without needing to compile OpenMP or another RAJA backend.

That being said, it would make sense for users just to use "plain" HiOp without RAJA in that case, as you shouldn't see any difference between CPU based HiOp, and RAJA with sequential policies.

Reply:
@cnpetra, using RAJA without OpenMP was a temporary workaround because of an issue on AMD architectures. There is no need to build RAJA without OpenMP backend anymore afaik.

Reply:
ok, good. Commit  https://github.com/LLNL/hiop/pull/543/commits/59466953fd65b5e233da789eabfcf7d9a5e355c7 removes `hiopVectorRajaPar`. Propagation of new template class went smoothly. But I think it is instructive to take a look at the commit's diffs, especially in the tests  since it points to some additional work that may be needed.

Reply:
> ok, good. Commit [5946695](https://github.com/LLNL/hiop/commit/59466953fd65b5e233da789eabfcf7d9a5e355c7) removes `hiopVectorRajaPar`. Propagation of new template class went smoothly. But I think it is instructive to take a look at the commit's diffs, especially in the tests since it points to some additional work that may be needed.

the commit also takes out the faulty preprocessor logic...